Remotion/Data/DomainObjects/Mapping/SortExpressions/SortExpressionParser.cs

```diff
@@ -135,10 +135,13 @@ private PropertyDefinition ParsePropertyName (string fullOrShortPropertyName)
       return propertyAccessorData.PropertyDefinition;
     }
 
-    private IEnumerable<ClassDefinition> GetAllDerivedClassDefinitions (ClassDefinition baseClassDefinition)
+    private IEnumerable<TypeDefinition> GetAllDerivedTypeDefinitions (TypeDefinition baseTypeDefinition)
     {
-      var directlyDerived = baseClassDefinition.DerivedClasses.Cast<ClassDefinition>();
-      return directlyDerived.Concat(directlyDerived.SelectMany(cd => GetAllDerivedClassDefinitions(cd)));
+      if (baseTypeDefinition is not ClassDefinition baseClassDefinition)
+        return Enumerable.Empty<TypeDefinition>();
+
+      var directlyDerived = baseClassDefinition.DerivedClasses.Cast<TypeDefinition>();
+      return directlyDerived.Concat(directlyDerived.SelectMany(cd => GetAllDerivedTypeDefinitions(cd))); // TODO R2I Mapping: decision needed if we want to support this for interfaces as well
```

Comment:
I think this has been ClassDefinition.GetAllDerivedTypes(), just manually implemented. Please check and if so, we want to change this to GetTypeHierarchy. The intention is to support properties available in the type hierarchy.

Update c# comments in line 105 and 117